example/hvstack.hpp

```diff
@@ -0,0 +1,95 @@
+#include "boost/gil/image_view_factory.hpp"
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <stdexcept>
+#include <vector>
+
+namespace boost { namespace gil {
+template <typename View>
+void hstack(const std::vector<View>& views, const View& output_view)
+{
+    if (views.size() == 0)
+    {
+        throw std::invalid_argument("empty views vector is passed - cannot create stacked image");
+    }
+
+    auto dimensions = views.front().dimensions();
+    for (const auto& view : views)
+    {
+        if (view.dimensions() != dimensions)
+        {
+            throw std::invalid_argument("one or many views are not of the same dimension");
+        }
+    }
+
+    auto full_width = dimensions.x * views.size();
+    if (output_view.width() != full_width || output_view.height() != dimensions.y)
+    {
+        throw std::invalid_argument("the destination view is not of the right dimensions");
+    }
+
+    for (std::size_t i = 0; i < views.size(); ++i)
+    {
+        auto subview = subimage_view(output_view, dimensions.x * i, 0, dimensions.x, dimensions.y);
+        copy_pixels(views[i], subview);
+    }
+}
+
+template <typename View>
+image<typename View::value_type> hstack(const std::vector<View>& views)
+{
+    if (views.size() == 0)
+    {
+        throw std::invalid_argument("empty views vector is passed - cannot create stacked image");
+    }
+
+    auto dimensions = views.front().dimensions();
+    image<typename View::value_type> result_image(dimensions.x * views.size(), dimensions.y);
```

Comment:
Also probably not a required feature, but maybe adding automatic padding to fit in the images could be a good idea. For example, relax the constraint of equal heights on the input images for vstack.

Reply:
I believe this is a bit more questionable compared to previous feature. Padding might make it hard to understand which images are padded which ones are not. Since users can pad themselves, I believe making it more explicit is the right way to go. If you have any more arguments in favor of the feature, please unresolve the conversation.

Reply:
@simmplecoder Yes, I get your argument.
My point was a simple one, if one provides image1 and image2 we could make a frame of dimension [max(image1.height(), image2.height()), max(image1.width(), image2.width())] for each of the images and stack these frames together. 
GIL already has code to enable padding along rows and columns which could come useful.
Anyways as I said it is not really a required feature, but just increases the flexibility of the stacking tool.

Reply:
@mloskot @lpranam what do you think? If the feature is not important from your point of view, the PR is ready for review.

Reply:
> GIL already has code to enable padding along rows and columns which could come useful.

True.

>  Since users can pad themselves, I believe making it more explicit is the right way to go.

Indeed. User has tools to compose desired processing chain.

Here, especially that this is still an example, I'd stick with the simplicity instead of offering rich set of options.